scipy/spatial/transform/rotation.py

```diff
@@ -1,12 +1,55 @@
 from __future__ import division, print_function, absolute_import
 
+__all__ = ['Rotation']
+
 import numpy as np
+import scipy.linalg
 
 class Rotation(object):
     """Rotation in 3 dimensions.
 
     This class will include initializers from different representations, converters
     and some useful algorithms such as quaternion slerp and rotation estimation.
+
     """
-    def __init__(self):
-        pass
+    def __init__(self, quaternions):
+        """Initialize class from possible normalized quaternions.
+
+        Parameters
+        ----------
+        quaternions : (N, 4) numpy.array
+                      Each row is a quaternion stored in scalar-last
+                      (x, y, z, w) format.
+
+        """
+        # Each row should have 4 numbers
+        assert(quaternions.shape[1] == 4)
+
+        # L2 norm of each row
+        norms = scipy.linalg.norm(quaternions, axis=1)
```

Comment:
I came to the conclusion that we need to pass a flag `normalize=True`. Meaning that if a user guarantees they are normalized we don't do that again. Makes sense to you?

Reply:
Sure, makes sense.
Just a semantic nitpick. How about a flag `normalized=True` flag? If the input quaternions are already `normalize`__d__, then we do not need to do  anything further.

Reply:
Haha, I thought about that too. Whatever feels more natural to you, if you choose `normalized`, then it should be `normalized=False`, i. e. by default we are cautious about that. Right?

Reply:
Yes, of course. If it is `normalized`, then default will be `False`. Sorry I forgot to clarify that.